### PR TITLE
inklecate: init at 1.0.0

### PR DESCRIPTION
**Pull request by shreerammodi**
###### Motivation for this change

The Inklecate compiler does not exist in the nixpkgs repo. I (and 2.8k stargazers) find the ink language useful so adding its official compiler is sure to help some nix users

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"` *-- No packages depend on this change*
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Shrimpram** [comment]
The error is `The option lib.maintainers.shreerammodi.name' is used but not defined`. Could be because that's the latter commit, I'll try reordering them

**Comment by Shrimpram** [comment]
Let me know if it's merge-able and I'll squash

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129591` at 95b6a047 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>inklecate</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/compilers/inklecate/default.nix:39:3:
```
   |
39 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> The error is `The option lib.maintainers.shreerammodi.name' is used but not defined`. Could be because the commit that updates the maintainers list is the latter commit, I'll try reordering them

That shouldn't matter.

File: pkgs/development/compilers/inklecate/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  src =
+    if stdenv.isLinux then
+      fetchurl {
+        url    = "${repo}/releases/download/v${version}/inklecate_linux.zip";
+        sha256 = "6e17db766222998ba0ae5a5da9857e34896e683b9ec42fad528c3f8bea7398ea";
+        name   = "${pname}-${version}";
+      }
+    else if stdenv.isDarwin then
+      fetchurl {
+        url    = "${repo}/releases/download/v${version}/inklecate_mac.zip";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/inklecate/default.nix:18]:
```suggestion
  src =
    if stdenv.isLinux then
      fetchurl {
        url    = "https://github.com/inkle/ink/releases/download/v${version}/inklecate_linux.zip";
        sha256 = "6e17db766222998ba0ae5a5da9857e34896e683b9ec42fad528c3f8bea7398ea";
        name   = "${pname}-${version}";
      }
    else if stdenv.isDarwin then
      fetchurl {
        url    = "https://github.com/inkle/ink/releases/download/v${version}/inklecate_mac.zip";
```

File: pkgs/development/compilers/inklecate/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    else throw "Not supported on ${stdenv.hostPlatform.system}.";
+
+  # Work around the "unpacker appears to have produced no directories"
+  # case that happens when the archive doesn't have a subdirectory.
+  setSourceRoot = "sourceRoot=$(pwd)";
+
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  phases = [ "unpackPhase" "installPhase" ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/inklecate/default.nix:31]:
```suggestion
```

disable phases which should not run instead.

File: pkgs/development/compilers/inklecate/default.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    rm $pname-$version/ink-engine-runtime.dll
+    rm $pname-$version/ink_compiler.dll
+    '';
+
+  installPhase = ''
+    mkdir -p $out/bin/
+
+    cp $pname-$version/inklecate $out/bin/inklecate
+    '';
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/inklecate/default.nix:43]:
```suggestion
  '';
```

File: pkgs/development/compilers/inklecate/default.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  phases = [ "unpackPhase" "installPhase" ];
+
+  unpackPhase = ''
+    unzip -qq -j $src -d $pname-$version
+
+    rm $pname-$version/ink-engine-runtime.dll
+    rm $pname-$version/ink_compiler.dll
+    '';
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/inklecate/default.nix:37]:
```suggestion
  '';
```

File: pkgs/development/compilers/inklecate/default.nix [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+
+  meta = with lib; {
+    description     = "Compiler for ink, inkle's scripting language";
+    longDescription = ''
+      Inklecate is a command-line compiler for ink, inkle's open source
+      scripting language for writing interactive narrative
+      '';
+    homepage        = "https://www.inklestudios.com/ink/";
+    downloadPage    = "${repo}/releases";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/inklecate/default.nix:53]:
```suggestion
    downloadPage    = "https://www.inklestudios.com/ink/";
```

File: pkgs/development/compilers/inklecate/default.nix [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    license         = lib.licenses.mit;
+    platforms       = platforms.linux ++ platforms.darwin;
+    maintainers     = with maintainers; [ shreerammodi ];
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/inklecate/default.nix:59]: One final new line to much.

File: pkgs/development/compilers/inklecate/default.nix [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  meta = with lib; {
+    description     = "Compiler for ink, inkle's scripting language";
+    longDescription = ''
+      Inklecate is a command-line compiler for ink, inkle's open source
+      scripting language for writing interactive narrative
+      '';
+    homepage        = "https://www.inklestudios.com/ink/";
+    downloadPage    = "${repo}/releases";
+    license         = lib.licenses.mit;
+    platforms       = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/inklecate/default.nix:55]:
```suggestion
    license         = licenses.mit;
    platforms       = platforms.unix;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/inklecate/default.nix (hunk 1 above)
Comment (Shrimpram) [reply pkgs/development/compilers/inklecate/default.nix]: Resolved by https://github.com/NixOS/nixpkgs/pull/129591/commits/7dd59fc311a3551925c745d605623e994bd08e9d

File: pkgs/development/compilers/inklecate/default.nix (hunk 6 above)
Comment (Shrimpram) [reply pkgs/development/compilers/inklecate/default.nix]: Screen Shot 2021-07-08 at 9 26 36 AM

**Comment by Shrimpram** [comment]
So sorry for all the errors, a lot of expressions currently in master use such code, which is what I based a lot of my expression off.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Cna you please squash all the package related commits together? make sure to leave the maintainer one extra.

**Pull request merged**